ARE/components/processor.adjustmentcurve/src/main/java/eu/asterics/component/processor/adjustmentcurve/AdjustmentCurveInstance.java

```diff
@@ -161,30 +187,52 @@ public double calculateOutputValue(double x) {
         CurvePoint a, b;
 
         if (curvePoints.size() > 1) {
-            a = curvePoints.get(0);
-            if (x <= a.x) {
-                return (a.y);
+            if (x <= getX(0)) {
+                return (getY(0));
             }
-            b = curvePoints.get(curvePoints.size() - 1);
-            if (x >= b.x) {
-                return (b.y);
+            if (x >= getX(curvePoints.size() - 1)) {
+                return (getY(curvePoints.size() - 1));
             }
             int i = 0;
-            while ((i < curvePoints.size() - 1) && (x > curvePoints.get(i).x)) {
+            while ((i < curvePoints.size() - 1) && (x > getX(i))) {
```

Comment:
some comments would be nice:
You are looking for the first point, where x > curvePoint.x(i)??

Reply:
Yes, looks like that this is what the code is doing... ;)
It's just the old logic where the x/y values now aren't directly retrieved form `curvePoints`, but using helper methods `getX()` and `getY()` for making a mode possible where the points are interpreted as percentage values. I'm sure the logic could be somehow improved/beautified but since there are no automatic tests I thought I'll better leave it as it is. However I did some manual tests with the test models existing for AdjustmentCurve and they worked the same way as before after my changes.